Fix bugs caused by html-safe empty strings

mongoose700 opened the pull request:
Adding empty strings as HTML leads to difficulties keeping track of
sibling nodes. It's easier to keep track of as text.

This fixes the issues https://github.com/emberjs/ember.js/issues/14924 and https://github.com/emberjs/ember.js/issues/16314. The first one could be solved by simply removing the `if (html === null || html === '')` special case. The latter issue cannot, because it would still need to deal with a `null` last node.

rwjblue commented:
Awesome, thank you for digging into this!

mongoose700 commented:
It seems to be failing some IE tests in CI, even though those tests don't even encounter my changes. Are they known to be flakey?

rwjblue commented:
I’ll restart, I’m not sure if they are normally flakey or not...

rwjblue commented:
I kicked it once more...

rwjblue commented:
Latest build had a single failure in IE11:

![screenshot](https://assets.saucelabs.com/jobs/1734150b98884e6e9b8c512911f1781e/0003screenshot.png)

mongoose700 commented:
How do you normally go about debugging issues that only happen in IE?

rwjblue commented:
The "easiest" thing would be to download an IE11 virtual machine (from [modern.ie](https://modern.ie)) and run the tests...

mongoose700 commented:
I've run the tests on a virtual machine repeatedly, and they never fail locally.

mongoose700 commented:
I've tried with Windows 10, and the test still passes. I switched to QUnit 2.0.0 to match what CI is using, and updated IE, which took me past the version used in CI. Is there a way I can use the exact same VM that CI is using?

[
<img width="1133" alt="screen shot 2018-03-13 at 12 52 20 am" src="https://user-images.githubusercontent.com/3057843/37328917-0351363e-2659-11e8-886e-a7ecd049003b.png">
](url)

chancancode commented:
@mongoose700 sorry this took so long! I merged in your changes in #852 and credited you for the commit. Would you mind opening a new PR to add the #each test you have here?
